### PR TITLE
[Documentation] Establish a project Code of Conduct

### PR DESCRIPTION
**Pull request by shatteredbeam**
This is a verbatim copy of the Contributor Covenant available here (https://www.contributor-covenant.org/version/1/4/code-of-conduct.html) and released via CC license.  The only modification made was to insert a proper email address.

I propose that we add this to the project, as we are currently lacking in a code of conduct.

### CODE REVIEW

**Review by Eearslya — APPROVED** [review]

**Review by theunkn0wn1 — APPROVED** [review]
LGTM

**Comment by 314numberpi** [comment]
[Question] Would the correct email not be ops?

**Comment by theunkn0wn1** [comment]
For issues arising from contribution guidelines & behavior on the repository,Is it notwithin the scope of the Project Manager (me) to handle such events? If the issue is sufficiently severe, I could escalate to the ops team. 

@kenneaal @UncleClapton Whats the ops team stance on this?

**Comment by kenneaal** [comment]
We actually discussed this in the ops time a while ago. When it comes to internal issues in the Tech Rats, we're keeping arbitration of such matters to ops who are also tech rats, since technical context is important in such matters.
By and large, project leads should seek to resolve any conflicts relating to their specific project, but may at any time request help or arbitration from ops. If project members feel that they have issues relating to a project lead that they don't want to bring up with them directly, they're free to contact us as well.

By and large, I hope that the need for a repo CoC is related to any external actors that may have interest in the project rather than internally in the tech team.

**Review by 314numberpi — APPROVED** [review]
Looks good, with the response to my question

File: CODE_OF_CONDUCT.md
Comment (theunkn0wn1) [inline CODE_OF_CONDUCT.md:53]: Just for the record, yeah this is too board for our purposes. i think @UncleClapton is doing something about it?

**Review by theunkn0wn1 — CHANGES_REQUESTED** [review]

File: CODE_OF_CONDUCT.md
Comment (UncleClapton) [reply CODE_OF_CONDUCT.md:53]: If I had to pick out what I disagree with **MOST**, it would be this line:

    Representation of a project may be further defined and clarified by project maintainers.

This opens things up for broad interpretation where broad interpretation **holds no place**. If we must adjust scope to fit something we're missing we can, but allowing individual maintainers to redefine where this CoC applies at-will is improper.

Since this CoC is intended for developers of Fuel Rats applications and services, a scope this wide is really overkill. While I certainly expect Crunchy and Xlexious to follow the idea of this CoC when tweeting (squeaking?) via our official Twitter, I don't think this specific CoC should bind their actions. The way it is currently worded, any activity from @fuelrats would be included, not just when we're speaking as techs. This is just one example of how the current scope is way too broad.

If I were to reword this to better fit our needs.. it might say something like... 

```
This Code of Conduct applies to everyone within the organization's project spaces,
and any individual officially representing the organization or project in public spaces.
```

However this alone is not enough as we need to further define the limit of what _"representing the organization"_ means. Otherwise it may be seen as _"as a maintainer you always represent the org and project."_ instead of _"as a maintainer you represent the org when speaking to users of the project, about the project"_.

Making that definition is something I don't feel comfortable fully defining on my own, and if we cannot get a stable definition then this CoC should be scoped back to just communication within projects. Personally I think it should be JUST the org and projects, but I'm willing to give applying a good expanded definition a shot if we can get good suggestions.

File: CODE_OF_CONDUCT.md
Comment (UncleClapton) [inline CODE_OF_CONDUCT.md:31]: This is highly up to personal interpretation IMO. Whats acceptable in my workplace may not be acceptable in yours. 

I realize I might be a bit pedantic about this one. I completely understand what the writers of the Contributor Covenant CoC are going for here, but it's the "walking the fine line" situations that may make some question rather this would apply to what they're saying, and I don't want to stifle conversation as a result.

In my experience.. if people question if what they're saying is acceptable (and I mean actually question it), they're more apt to just not say anything at all instead of asking for clarification. I'm not saying all people are like this.. but there are quite a few.

For that reason I'm very opposed to overly-broad and easily misinterpreted wording such as this.

File: CODE_OF_CONDUCT.md
Comment (UncleClapton) [inline CODE_OF_CONDUCT.md:42]: Unfortunately this text does not work well with the split jurisdiction we're holding on org repos. This will probably have to get re-worded eventually to specify what maintainers are responsible for and what tech-mods are responsible for.

Lets hold that off until we're sure how we're moving forward with the adoption of this CoC, though.

File: CODE_OF_CONDUCT.md
Comment (UncleClapton) [inline CODE_OF_CONDUCT.md:58]: This is something we're gonna have to re-address as well.

I think actual reports of abusive behavior coming from outside collaborators should go to the "tech rat moderators" (really don't have a good name for those who exist in the overlap), as their complains are more likely to be of someone from within the org.

We also need to make a final decision on how bans of those outside of the tech rats will work... I believe disagreements and conflicts should be settled with conversation whenever possible (I feel like CMDR Rhea advocating against discord's partnership CoC :P), but if a project lead thinks that a ban is **absolutely necessary** then the case should be handed over to the "techs who are also mods" group.

I think this step is necessary because anyone who has broken the CoC that badly here, should also be considered for an IRC ban or FR account suspension. This cannot happen if us mods can't investigate and discuss the right course of action on our own.

File: CODE_OF_CONDUCT.md
Comment (UncleClapton) [inline CODE_OF_CONDUCT.md:8]: 💢 This list doesn't include political views. Why does it not include political views?

**Review by UncleClapton — CHANGES_REQUESTED** [review]
So the idea I proposed way back when this started was if we all (as in us tech rats) found it suitable, we should apply this to the whole org instead of just this project. If we go down that route, then this PR should be closed, and the text transferred to a separate repo. Probably [tech-rat-code-standards](https://github.com/FuelRats/tech-rat-code-standards), though we will rename that repo before this is put into effect.

In this specific document's place, A new file should be made containing a link to the org-wide CoC, and quick contact links for complaints and questions.

I think some additional discussion on that point should be made before we close this PR, however.

Finally, I realize a lot of my complaints in this review stem from my general feelings of this particular CoC template. Let me be clear when I say I ***HATE*** the Contributor Covenant. Up front it looks very solid and takes hard stances on whats acceptable, but when read into it's clear that as-is it leaves gaping holes for interpretation to reject contributions for anything the maintainers might disagree with, while giving the appearance that they're pro acceptance and pro free speech. What sucks is that it's easily fixable, but often just left as-is with the glaring holes in tact...

**Comment by shatteredbeam** [comment]
Closed.  This discussion is being moved to an org-wide level to be discussed with that scope in mind.

**Pull request closed without merging**
